source/image/ImageBuffer.cpp

```diff
@@ -308,7 +311,7 @@ namespace {
 		jpeg_create_decompress(&cinfo);
 #pragma GCC diagnostic pop
 
-		jpeg_stdio_src(&cinfo, file);
+		jpeg_mem_src(&cinfo, reinterpret_cast<const unsigned char *>(data.c_str()), data.size() + 1);
```

Comment:
Why the +1?
```suggestion
		jpeg_mem_src(&cinfo, reinterpret_cast<const unsigned char *>(data.c_str()), data.size());
```

Reply:
I think that's for the null terminating byte? I'll check.

Reply:
Yeah. It's not needed for it to work, but that's technically the size of the memory buffer that we have.

Reply:
So if we don't actually need the nullterm, we can just do this:
```suggestion
		jpeg_mem_src(&cinfo, reinterpret_cast<const unsigned char *>(data.data()), data.size());
```